### PR TITLE
Get episodes by period#3095

### PR DESCRIPTION
**Pull request by marinakr**

### CODE REVIEW

File: apps/api/test/api_web/controllers/episode_controller_test.exs [hunk 1]
```diff
@@ -506,7 +506,7 @@ defmodule Api.Web.EpisodeControllerTest do
     end
   end
 
-  describe "list episodes" do
+  describe "list episodes any episodes" do
```
[the remark this answers is not in the document]
Comment (marinakr) [reply apps/api/test/api_web/controllers/episode_controller_test.exs]: list any episodes without any search parameters

File: apps/core/lib/core/patients/episodes/episodes.ex [hunk 2]
```diff
@@ -42,26 +42,44 @@ defmodule Core.Patients.Episodes do
...
+          }
+        },
+        %{"$match" => %{"period_match" => true}},
+        %{"$project" => %{"period_match" => 0}}
       ]
   end
 
-  defp period_compare_function("period_from"), do: {"$gte", "$period.start"}
-  defp period_compare_function("period_to"), do: {"$lte", "$period.end"}
+  defp period_compare_function("period_from", date),
```
Comment (AlexKovalevych) [inline apps/core/lib/core/patients/episodes/episodes.ex]: please use full function definition if it's not simple (doesn't fit in a single line):

```
def ... do
end
```

File: apps/core/lib/core/patients/episodes/episodes.ex [hunk 3]
```diff
@@ -42,26 +42,44 @@ defmodule Core.Patients.Episodes do
...
+    do: %{
+      "$or" => [
+        %{"$gte" => ["$period.start", date]},
+        %{"$gte" => ["$period.end", date]},
+        %{"$eq" => ["$period.end", nil]}
+      ]
+    }
+
+  defp period_compare_function("period_to", date),
```
Comment (AlexKovalevych) [inline apps/core/lib/core/patients/episodes/episodes.ex]: same here

File: apps/core/lib/core/patients/episodes/episodes.ex [hunk 4]
```diff
@@ -42,26 +42,44 @@ defmodule Core.Patients.Episodes do
...
+        %{"$project" => %{"period_match" => 0}}
       ]
   end
 
-  defp period_compare_function("period_from"), do: {"$gte", "$period.start"}
-  defp period_compare_function("period_to"), do: {"$lte", "$period.end"}
+  defp period_compare_function("period_from", date),
+    do: %{
+      "$or" => [
+        %{"$gte" => ["$period.start", date]},
```
Comment (AlexKovalevych) [inline apps/core/lib/core/patients/episodes/episodes.ex]: are you sure about these conditions? they look weird

**Pull request merged**